### ISSUE TITLE
Task: Reorganize Go Project Structure

### ISSUE THREAD

**Issue by zdylag** [speedrun-website/speedrun-rest#38]
Right now we have an app directory and a few others containing go code. This does not seem to be industry standard, so let's move to something more common.

Example apps (benchstat) seem to use a cmd directory to contain their apps.

This is recommended by https://github.com/golang-standards/project-layout which has some other recommendations as well, which this task can optionally try to follow.

### Linked Issues
None

### Affected Components
None

**Comment by FrozenSake** [comment]
One comment I'd make here is that that standard is much maligned and contentious. [This ](https://github.com/golang-standards/project-layout/issues/117) is not a fun read (unless you like popcorn drama), but I think shows the depth of the contention.

**Comment by zdylag** [comment]
Yikes, yeah the pkg part in particular did stand out as sounding less like most of the main go tools. 

@FrozenSake Since your comment is now here I am leaning towards leaving the description as is, since the main move is from app->cmd.

Does that seem reasonable?

**Comment by FrozenSake** [comment]
Yeah, I think that's a good one. I'm not the main maintainer on my company's go monolith, but I was planning to ask him for some advice tomorrow morning as well - since saying "this is a bad suggestion" without having anything to provide in response isn't helpful.

I think the cmd pattern is super common and makes sense. And my take on the spirit of go is "do what makes sense" haha.

**Comment by FrozenSake** [comment]
Okay, got advice from him, which was:
> Rather than cookie-cutter project structure, I like thinking about "what goes in a package". Since the directory hierarachy means nothing in Go, nesting packages is just a matter of taste, basically. Here are the articles I've liked about package structure:

- https://www.gobeyond.dev/standard-package-layout/ and https://www.gobeyond.dev/packages-as-layers/
- https://www.ardanlabs.com/blog/2017/02/package-oriented-design.html and https://www.ardanlabs.com/blog/2017/02/design-philosophy-on-packaging.html
- https://bencane.com/stories/2020/07/06/how-i-structure-go-packages/ (this one's great!)
- https://rakyll.org/style-packages/

> Ah, and "[Effective Go](https://golang.org/doc/effective_go)" and "[Go Code Review](https://github.com/golang/go/wiki/CodeReviewComments)" comments as overriding guidelines... such as "no underscore in package names", and "make the directory name and the package name always the same"

I'm still digesting all of them and trying to think about what that means structure wise, but I think some basic stuff is usable there, and the move to app -> cmd makes sense to me as well. Just didn't wanna leave this hanging at "I'll ask a guy"
